datafusion/src/scalar.rs

```diff
@@ -526,6 +526,282 @@ macro_rules! eq_array_primitive {
 }
 
 impl ScalarValue {
+    /// Return true if the value is numeric
+    pub fn is_numeric(&self) -> bool {
+        matches!(self,
+            ScalarValue::Float32(_)
+            | ScalarValue::Float64(_)
+            | ScalarValue::Decimal128(_, _, _)
+            | ScalarValue::Int8(_)
+            | ScalarValue::Int16(_)
+            | ScalarValue::Int32(_)
+            | ScalarValue::Int64(_)
+            | ScalarValue::UInt8(_)
+            | ScalarValue::UInt16(_)
+            | ScalarValue::UInt32(_)
+            | ScalarValue::UInt64(_)
+        )
+    }
+
+    /// Add two numeric ScalarValues
+    pub fn add(lhs: &ScalarValue, rhs: &ScalarValue) -> Result<ScalarValue> {
```

Comment:
Completely agree. My feeling is these basic operations could be implemented in ScalarValue since many operators will depend on them. I want to pick your brain (and other contributor's too) if you think this is a good idea - this deserves a seperate discussion. Also with the way ScalarValue is implemented currently each operator needs a high match block, maybe using generics can make it a little easier? 

Here are some options I propose:
1. Remove the redundency, I can either move sum to Scalavalue or the other direction is fine. Personally I prefer to have it in ScalarValue.
2. Leave the code as is until we have a discussion/conclusion around how to handle arithmetic operation for ScalarValue.

Reply:
The different value from postgres and google is because of the nature of stddev, there are two versions: 1. population and 2. sample. The one in this PR is population, looks like the default one with postgres and google is sample. The difference in the calculation is very minimal I can include the sampe version in the PR as well. Good catch! 

So my proposal is to add two functions: stddev_pop and stddev_s, does this look reasonable? @alamb 

And I will look into the inconsistency between query runs. Thanks!

Reply:
The inconsistency between query runs are now fixed.

Reply:
yes that sounds good. Thank you